### PR TITLE
Add Https to HttpIppClientTransport sample

### PR DESCRIPTION
**Pull request by psoreide**
A new option has been added to jprint: -s / --self-signed to accept self signed certificates

Signed-off-by: PK Søreide <per.kristian.soreide@gmail.com>

### CODE REVIEW

File: sample/jprint/src/main/java/sample/HttpIppClientTransport.java [hunk 1]
```diff
@@ -61,4 +82,53 @@ private void copy(InputStream data, OutputStream output) throws IOException {
...
+
+            SSLContext sslContext = SSLContext.getInstance(SSL_PROTOCOL);
+            sslContext.init(null, TRUST_ALL_CERTS, new java.security.SecureRandom());
+            return sslContext;
+        } catch (NoSuchAlgorithmException | KeyManagementException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static final TrustManager[] TRUST_ALL_CERTS;
```
Comment (GladeDiviney) [inline sample/jprint/src/main/java/sample/HttpIppClientTransport.java:112]: let's keep all statics at top of file

File: sample/jprint/src/main/java/sample/HttpIppClientTransport.java [hunk 2]
```diff
@@ -12,20 +12,41 @@
...
  *
  * It is assumed that the remote server will not deliver additional data (just an IPP packet).
  */
 class HttpIppClientTransport implements IppClientTransport {
+    private static final String SSL_PROTOCOL = "TLSv1.2";
+
+    private final SSLContext sslContext;
+	private final boolean acceptSelfSignedCerts;
+
+    HttpIppClientTransport(boolean acceptSelfSignedCerts) {
```
Comment (GladeDiviney) [inline sample/jprint/src/main/java/sample/HttpIppClientTransport.java:37]: Let's add comment:

```suggestion
    /**
     * @param acceptSelfSignedCerts If true, auto-accept self-signed HTTPS certificates. Real implementations should
     *   implement a Trust-On-First-Use approach to minimize the potential for MITM attacks.
     */
     HttpIppClientTransport(boolean acceptSelfSignedCerts) {
```

File: sample/jprint/src/main/java/sample/Main.java [hunk 3]
```diff
@@ -67,6 +68,12 @@ public static void main(String[] args) throws IOException {
                 help();
             }
 
+            if (command.hasOption("s")) {
+                transport = new HttpIppClientTransport(true);
+            } else {
+                transport = new HttpIppClientTransport(false);
+            }
```
Comment (GladeDiviney) [inline sample/jprint/src/main/java/sample/Main.java:75]: Can be
```suggestion
            transport = new HttpIppClientTransport(command.hasOption("s"));
```

**Review by GladeDiviney — COMMENTED** [review]
Thanks for the submission! Minor comments.

**Comment by psoreide** [comment]
Hey! Looks like the javadoc ended up on the wrong constructor!

**Pull request merged**